service/src/main/kotlin/app/cash/backfila/service/DbBackfillRun.kt

```diff
@@ -12,43 +21,34 @@ import javax.persistence.JoinColumn
 import javax.persistence.ManyToOne
 import javax.persistence.Table
 import javax.persistence.Version
-import misk.hibernate.DbTimestampedEntity
-import misk.hibernate.DbUnsharded
-import misk.hibernate.Id
-import misk.hibernate.JsonColumn
-import misk.hibernate.Query
-import misk.hibernate.Session
-import misk.hibernate.newQuery
-import okio.ByteString
-import okio.ByteString.Companion.decodeBase64
 
 /**
  * Tracks the state of a created backfill.
  */
 @Entity
 @Table(name = "backfill_runs")
-class DbBackfillRun() : DbUnsharded<DbBackfillRun>, DbTimestampedEntity {
+open class DbBackfillRun() : DbUnsharded<DbBackfillRun>, DbTimestampedEntity {
```

Comment:
all the Db classes have to be non-final so hibernate can proxy them!

Reply:
For the paginator I think yeah?

Reply:
it's not necessary for any of this to work, but what happens if hibernate can't proxy is it loads all the join columns eagerly, even if you don't use them. So when loading a DbBackfillRun it loaded the DbRegisteredBackfill each time causing n queries

Reply:
Do the members have to be open too or just the class?